ambry-server/src/main/java/com/github/ambry/server/RepairRequestsSender.java

```diff
@@ -0,0 +1,237 @@
+        Thread.sleep(SLEEP_TIME_MS);
+
+        // loop all the partitions
+        for (long partitionId : partitionIds) {
+          if (shutdown) {
+            return;
+          }
+
+          // get the repair requests for this partition
+          requestInfos = getAmbryRequest(partitionId);
+          if (requestInfos == null || requestInfos.isEmpty()) {
+            continue;
+          }
+
+          // send the repair requests to the handler and wait for all the responses.
+          Set<Integer> requestsToDrop = new HashSet<>();
+          int inflightReqCount = requestInfos.size();
+          logger.info("RepairRequests Sender: {} sending {} repair requests ", nodeId, inflightReqCount);
+          List<ResponseInfo> responses = client.sendAndPoll(requestInfos, requestsToDrop, POLL_TIMEOUT_MS);
+          requestInfos.clear();
+
+          for (ResponseInfo info : responses) {
+            inflightReqCount--;
+            ReplicateBlobResponse res = (ReplicateBlobResponse) info.getResponse();
+            ReplicateBlobRequest req = (ReplicateBlobRequest) info.getRequestInfo().getRequest();
+            if (res.getError() == ServerErrorCode.No_Error) {
+              RepairRequestRecord.OperationType type;
+              if (req.getOperationType() == RequestOrResponseType.TtlUpdateRequest) {
+                type = RepairRequestRecord.OperationType.TtlUpdateRequest;
+              } else {
+                type = RepairRequestRecord.OperationType.DeleteRequest;
+              }
+              db.removeRepairRequests(req.getBlobId().toString(), type);
+              logger.info("RepairRequests Sender: Repaired {}", req);
+            } else {
+              // LOCAL_CONSISTENCY_TODO: metrics for all the errors happens in RepairRequestsSender.
+              // Ideally we may need create alert if we cannot repair the requests in timely matter.
+              logger.info("RepairRequests Sender: failed to repair {} response {} ", req, res);
+            }
+          }
+
+          if (inflightReqCount > 0) {
+            logger.error("RepairRequests Sender: timeout waiting for repairs {} {}", nodeId, inflightReqCount);
+          }
+        } // loop all the partitions
+      } catch (Throwable e) {
+        // TODO add metric to track background threads
+        logger.error("RepairRequests Sender: Exception when handling request", e);
+        // this is bad and we need to shutdown the app
+        Runtime.getRuntime().halt(1);
+      } finally {
+        if (requestInfos != null) {
+          for (RequestInfo reqInfo : requestInfos) {
+            reqInfo.getRequest().release();
+          }
+          requestInfos = null;
+        }
+      }
+    }
+  }
+
+  /**
+   * Get the {@link RepairRequestRecord} from the database and generate {@link ReplicateBlobRequest}
+   * @param partitionId the partition id
+   * @return list of {@link RequestInfo}
+   */
+  private List<RequestInfo> getAmbryRequest(long partitionId) throws Exception {
```

Comment:
nit: can we rename this method to something like getRepairRequestsForPartition?

Reply:
Renamed accordingly.

Reply:
I don't think you renamed?